### PR TITLE
feat(relay): Expose event retention to Relay

### PR DESCRIPTION
**Pull request by jan-auer**
This allows relay to set the retention directly for session events.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py [hunk 1]
```diff
@@ -150,6 +151,12 @@ def test_internal_relays_should_receive_full_configs(
     assert safe.get_path(cfg, "config", "datascrubbingSettings", "scrubIpAddresses") is True
     assert safe.get_path(cfg, "config", "datascrubbingSettings", "sensitiveFields") == []
 
+    # Event retention depends on settings, so assert the actual value. Likely
+    # `None` in dev, but must not be missing.
+    assert cfg["config"]["eventRetention"] == quotas.get_event_retention(
```
Comment (untitaker) [inline tests/sentry/api/endpoints/test_relay_projectconfigs.py]: Can we just hardcode the value here? We know what our test settings look like.

**Review by untitaker — APPROVED** [review]

File: tests/sentry/api/endpoints/test_relay_projectconfigs.py (hunk 1 above)
Comment (jan-auer) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py]: The hard coded one is None ^^

Comment (untitaker) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py]: ok can we set the event retention to something and assert it works?

Comment (jan-auer) [reply tests/sentry/api/endpoints/test_relay_projectconfigs.py]: Potentially, with a global option. The actual code where this is set lives outside of sentry, so I didn't see a point in this.

**Pull request merged**